operator/controllers/helmrelease_controller.go

```diff
@@ -44,19 +59,33 @@ func (r *HelmReleaseReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	if err := r.Get(ctx, req.NamespacedName, &helmRelease); err != nil {
 		if apierrs.IsNotFound(err) {
 			log.Info("HelmRelease doesn't exist")
-			return ctrl.Result{}, nil
+			return ctrl.Result{}, r.onDelete(ctx, req.NamespacedName)
 		}
 
 		return ctrl.Result{}, err
 	}
 
-	log.Info("Make things happen", "test", helmRelease.Spec)
-
-	return ctrl.Result{}, nil
+	return ctrl.Result{}, r.onUpdate(ctx, helmRelease)
 }
 
 func (r *HelmReleaseReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&shipitv1beta1.HelmRelease{}).
 		Complete(r)
 }
+
+func (r *HelmReleaseReconciler) onDelete(ctx context.Context, name types.NamespacedName) error {
+	// Update HelmRelease Status to 'PENDING_DELETE' -- but the HelmRelease doesn't exist anymore :(
```

Comment:
Not sure what we'd do here to reflect the `PENDING_DELETE` state, the HelmRelease CR was already deleted by this point

Reply:
As we discussed this morning, if the cluster is in the middle of deleting the release, perhaps we don't really have a need to observe this particular state.